src/lib/mina/precondition.ts

```diff
@@ -364,13 +366,16 @@ function preconditionSubclass<
         longKey
       ) as AnyCondition<U>;
       if ('isSome' in property) {
-        property.isSome = Bool(true);
-        if ('lower' in property.value && 'upper' in property.value) {
-          property.value.lower = value;
-          property.value.upper = value;
+        let isInterval = 'lower' in property.value && 'upper' in property.value;
+        let newValue = isInterval ? { lower: value, upper: value } : value;
+        ensureConsistentPrecondition(property, newValue);
+        if (isInterval) {
+          (property.value as { lower: U; upper: U }).lower = value;
+          (property.value as { lower: U; upper: U }).upper = value;
         } else {
           property.value = value;
         }
```

Comment:
could be just this right?
```suggestion
        property.value = newValue;
```